### PR TITLE
Bower to npm

### PR DESCRIPTION
**Pull request by nschonni**
Related to #8063 but there are a few left over that will take some more effort. 

They don't have official NPM packages:
- [ ] Flot
- [ ] excanvas
- [ ] SideBySideImproved

There is a fork/commit being used:
- [x] magnific-popup
- [ ] openlayers

This could land as-is, but probably needs a good review to see it I overlooked anything when repointing stuff.

There is also a few dependies that are just dropped, since they were orphans from the v4 alpha stage

### CODE REVIEW

**Comment by LaurentGoderre** [comment]
@nschonni can you also remove the lock file. Since we fix the node issue we shouldn't pin everything back like that

**Comment by nschonni** [comment]
@LaurentGoderre the NPM 5 lock file is a little different in that it actually works :wink: 
I could remove it, but since we have a bunch of pinned dependencies, I think it makes sense to leave it until everything can move to a `^` dependency

**Comment by LaurentGoderre** [comment]
Ok!

**Comment by duboisp** [comment]
@nschonni can you fix the Conflicting file

Thanks

File: Gruntfile.coffee
Comment (nschonni) [inline Gruntfile.coffee]: Oops, need to fix whitespace in this file again since I think the MathJax added that accidentally

**Comment by nschonni** [comment]
This has been updated to include the MathJax changes and Magnific

**Comment by lucas-hay** [comment]
@nschonni is this ready to merge now?

**Comment by nschonni** [comment]
Yes

**Comment by lucas-hay** [comment]
@LaurentGoderre are you OK with this being merged?

File: bower.json
Comment (LaurentGoderre) [inline bower.json]: Why can't these two go in the `package.json`?

Comment (nschonni) [reply bower.json]: NPM can only use a Git source if the repo has a package.json in the repo. Those currently don't

Comment (LaurentGoderre) [reply bower.json]: OOh!

Comment (LaurentGoderre) [reply bower.json]: Well openlayers we kind of control so you can create a PR there for a package.json

File: bower.json
Comment (LaurentGoderre) [inline bower.json]: We can fork this one and add a package.json. We wont need this in the future of WET so we're just keeping it to not breaking existing things.

File: bower.json
Comment (nschonni) [reply bower.json]: Yeah, there is the separate PR to try and update OpenLayers, so I didn't want to cause more conflicts. I think newer version of the library are just published normally to NPM

File: bower.json
Comment (nschonni) [reply bower.json]: Yup, but that can be done then updated in a separate PR

Comment (LaurentGoderre) [reply bower.json]: I agree!

File: bower.json
Comment (jvanulde) [reply bower.json]: OpenLayers is coming via NPM now in the latest PR.

**Comment by nschonni** [comment]
@LaurentGoderre do you want to land this?

**Comment by duboisp** [comment]
@lucas-hay see @nschonni [comments](https://github.com/wet-boew/wet-boew/pull/8147#issuecomment-349151784)

**Comment by LaurentGoderre** [comment]
I don't have time to test irt but if you're confident, I don't see why not!

**Review by LaurentGoderre — APPROVED** [review]

**Comment by LaurentGoderre** [comment]
@nschonni do you want a squash? I think we should merge multiple commits for this one.

**Comment by nschonni** [comment]
No, I rebased it so each of the commits is pretty atomic, so I don't think the squash is needed

**Comment by duboisp** [comment]
@lucas-hay I think this is the PR that are currently breaking the GCWeb build.

@nschonni @LaurentGoderre can this PR be reverted for now and be merge at the next PR review meeting?

**Comment by lucas-hay** [comment]
@nschonni @LaurentGoderre Can this be fixed in GCWEB or reverted here?
I will finally be releasing 4.0.27 tomorrow and my schedule can't afford any delays tomorrow. I would assume that perhaps the other repo's may have the same problem.

**Comment by EricDunsworth** [comment]
@lucas-hay @duboisp I haven't tested this myself, but wet-boew/GCWeb#1320's [Travis-CI error](https://travis-ci.org/wet-boew/GCWeb/builds/312010980#L245) can probably be resolved by replicating one of [this PR's Gruntfile changes](https://github.com/wet-boew/wet-boew/pull/8147/files#diff-a35864fcc4749a3b904641b4be9feacfR313) in [GCWeb's Gruntfile](https://github.com/wet-boew/GCWeb/blob/master/Gruntfile.coffee#L187).

I could send over a GCWeb PR with that change if needed. But I don't have time to test it locally and thus can't guarantee that there won't be any further errors past that point in the build process.

**Comment by lucas-hay** [comment]
@EricDunsworth We would likely need to do this with all repos and with the release tomorrow I think the best bet is to revert for now and we can merge after release is complete and then check other repos.

**Comment by EricDunsworth** [comment]
@duboisp @lucas-hay Are there any plans to re-implement this PR's changes (since #8228 reverted it)?

I personally would've preferred quick-fixing the other themes rather than reverting this PR. The changes it brought in were valid and didn't break this repo's build system.

IMO it would've been more ideal to fix the other themes' build systems - even if it resulted in 4.0.27 getting further delayed. Plus in the end, due to other issues, 4.0.27 didn't end up getting released until over a week after #8228 was merged in.

**Comment by duboisp** [comment]
@EricDunsworth The first thing will be to have all those PR's ready to be merge. Then, after testing, we would be able to merge it and re-apply the change from this PR.

Feel free to submit all those related PR's then it can be merged all at once during a "PR review" meeting.

**Comment by LaurentGoderre** [comment]
Testing this can't be done online btw. You would have to run a local npm link

**Pull request merged**
